### PR TITLE
[frontend] Fix Adding a tag to a dashboard is not working (#4863)

### PR DESCRIPTION
**Pull request by SarahBocognano**
### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/4863

### CODE REVIEW

**Comment by helene-nguyen** [comment]
Adding one tag works fine :) But when adding multiple tags, the `MORE...` button appears and when click on it, nothing happens

image

It would be great to align the `WORK...` button with the others :)

What do you mean by the `WORK...` button ?  @helene-nguyen

**Comment by Archidoit** [comment]
When I add several tags, there is an overlap: 
image

But if I add enough tags, there is no more overlap:  
image


@Archidoit Yep I'm trying to work on it. Since OpenCTI is not responsive it's overlapping on those one -_-'

**Comment by SarahBocognano** [comment]
Update: After a talk with the product team. Knowing that the app is not really responsive, we've decided to put the tags under the buttons. All comments have been resolved

**Comment by SamuelHassine** [comment]
Adding more than 5 tags displays the "More" button which is not working in this branch.

Also, I'm not sure about putting the tags below, which is not so user friendly for dashboards. I would prefer we hide any overflow on the line, even if only one tag is displayed. And have a button to open the rest.

**Comment by SarahBocognano** [comment]
Update: style problem fix, and I align the "More" tags button with the ones in aliases in entities

**Comment by helene-nguyen** [comment]
Canwe add more than 5 tags? The 'plus' button disappears and the 'more' button doesn't work

image

@helene-nguyen  You're not supposed to have this visual. The more button is not there anymore, it is manage like the aliases.

**Comment by SamuelHassine** [comment]
And the "More" button should have a margin right to avoid to be closed to the orange button. Also alignment is not correct we should margin top negative to align with current right buttons position.

@SamuelHassine  It's not a more button anymore, it is manage like the aliases. I'm not sure that the comment up matter because it's not align with the last version of this PR.

**Comment by Kedae** [comment]
image
Aliases or tag ?

File: opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.jsx [hunk 1]
```diff
@@ -102,52 +117,68 @@ const WorkspaceHeader = ({
...
+  const openTagsCreate = false;
   const [openTag, setOpenTag] = useState(false);
+  const [newTag, setNewTag] = useState('');
   const [openTags, setOpenTags] = useState(false);
   const userCanManage = workspace.currentUserAccessRight === 'admin';
   const userCanEdit = userCanManage || workspace.currentUserAccessRight === 'edit';
   const [displayDuplicate, setDisplayDuplicate] = useState(false);
   const handleCloseDuplicate = () => setDisplayDuplicate(false);
   const [duplicating, setDuplicating] = useState(false);
+  const tags = R.propOr([], 'tags', workspace);
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.jsx]:
```suggestion
  const tags = workspace && workspace.tags ? workspace.tags : [];
```

File: opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.jsx [hunk 2]
```diff
@@ -102,52 +117,68 @@ const WorkspaceHeader = ({
...
+        && newTag !== ''
     ) {
       commitMutation({
         mutation: workspaceMutation,
         variables: {
           id: workspace.id,
           input: {
             key: 'tags',
-            value: R.append(data.new_tag, currentTags),
+            value: R.append(newTag, currentTags),
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceHeader.jsx]:
```suggestion
            value: [...currentTags, newTag],

```

**Review by marieflorescontact — COMMENTED** [review]
locally tested 👌

**Comment by helene-nguyen** [comment]
Locally tested too, everything seems good :)

**Review by helene-nguyen — APPROVED** [review]

**Pull request merged**